src/string-calculator.ts

```diff
@@ -1,12 +1,235 @@
+        if (this.numberList.includes("//")) {
+            let sep: string = this.getCustomSeparator();
+            res = _.escapeRegExp(sep);
+            this.customSep = sep;
+            this.numberList = this.numberList.substring(sep.length + 3); // remove custom separator from number list
+        }
+
+        return res;
+    }
+
+    private getCustomSeparator(): string {
+        let beginOfSep: number = this.numberList.indexOf("//") + 2;
+        let endOfSep: number = this.numberList.indexOf("\n");
+        if (beginOfSep != -1) {
+            return this.numberList.substring(beginOfSep, endOfSep);
+        }
+
         return "";
     }
-    
+
+    private checkForErrors(number: string):ErrorCode {
+        let err: ErrorCode;
+        
+        err = this.isNumberNegative(number);
+        if(err) {
+            return err;
+        }
+        err = this.isNumberMissed(number);
+        if(err) {
+            return err;
+        }
+        err = this.isNumberNumeric(number);
+        if(err) {
+            return err;
+        }
+        return null;
+    }
+
+    private isNumberNegative(number: string): ErrorCode {
+        if (number.startsWith("-")) {
+            //throw new NegativeNumberException();
+            return ErrorCode.NegativeNumberErrorCode;
+        }
+        return null;
+    }
+
+    private isNumberMissed(number: string) {
+        if (number.trim().length == 0) {
+            if (this.numberList.includes(",,") || this.numberList.includes("\n,")) {
+                //throw new CommaFoundException();
+                return ErrorCode.CommaFoundErrorCode;
+            }
+            if (this.numberList.includes(",\n") || this.numberList.includes("\n\n")) {
+                //throw new EndOfLineFoundException();
+                return ErrorCode.EndOfLineFoundErrorCode;
+            }
+            if (this.customSep && this.numberList.includes(this.customSep.concat(this.customSep))) {
+                //throw new CustomSeparatorFoundException();
+                return ErrorCode.CustomSeparatorFoundErrorCode;
+            }
+            if (this.numberList.endsWith(",") || this.numberList.endsWith("\n") || (this.numberList.endsWith(this.customSep) && this.customSep.trim().length != 0)) {
+                //throw new LastNumberMissedException();
+                return ErrorCode.LastNumberMissedErrorCode;
+            }
+        }
+
+        return null;
+    }
+
+    private isNumberNumeric(number: string) {
+        if (!(/^\d+?(\.\d+)?$/.test(number))) {
+            //throw new NumberNotNumericException();
+            return ErrorCode.NumberNotNumericErrorCode;
+        }
+
+        return null;
+    }
+
+    private getIndexOfUnexpectedComma(): number {
+        if (this.numberList.includes(",,")) {
+            return this.numberList.indexOf(",,") + 1;
+        }
+
+        return this.numberList.indexOf("\n,") + 1;
+    }
+
+    private getIndexOfUnexpectedCustomSeparator(): number {
```

Comment:
how are getIndexOfUnexpectedComma and getIndexOfUnexpectedCustomSeparator different?
I'd say the rule does not allow 2 subsequent separators:

- by default: ,, | ,\n | \n\n
- custom: sepsep
- (this one we might argue about) custom: sep\n | \n\n 

we should consider new lines as alternative separator:
'Newline as separator"